fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/LaoDetailViewModel.java

```diff
@@ -354,4 +348,64 @@ public void subscribeToLao(String laoId) {
                   mIsOrganizer.postValue(false);
                 }));
   }
+
+  public void setCurrentLaoName(String laoName) {
+    if (laoName != null && !laoName.isEmpty() && !laoName.equals(getCurrentLaoName())) {
+      Log.d(TAG, "New name for current LAO: " + laoName);
+      mLaoName.setValue(laoName);
+    }
+  }
+
+
+  public void createNewElection(String name, long start, long end, String votingMethod, boolean writeIn, List<String> ballotOptions, String question) {
+    Log.d(TAG,"creating a new election with name " + name);
+
+    Lao lao = getCurrentLao();
+    if (lao == null) {
+      Log.d(TAG, "failed to retrieve current lao");
+      return;
+    }
+
+    String channel = lao.getChannel();
+    ElectionSetup electionSetup;
+    String laoId = channel.substring(6);
+
+    electionSetup = new ElectionSetup(name, start, end, votingMethod, writeIn, ballotOptions, question, laoId);
+
+    try {
+      // Retrieve identity of who is creating the election
+      KeysetHandle publicKeysetHandle = mKeysetManager.getKeysetHandle().getPublicKeysetHandle();
+      String publicKey = Keys.getEncodedKey(publicKeysetHandle);
+      byte[] sender = Base64.getDecoder().decode(publicKey);
+      PublicKeySign signer = mKeysetManager.getKeysetHandle().getPrimitive(PublicKeySign.class);
+      MessageGeneral msg = new MessageGeneral(sender, electionSetup, signer, mGson);
+
+      Log.d(TAG, "sending publish message");
+      Disposable disposable =
+              mLAORepository
+                      .sendPublish(channel, msg)
+                      .subscribeOn(Schedulers.io())
+                      .observeOn(AndroidSchedulers.mainThread())
+                      .timeout(5, TimeUnit.SECONDS)
+                      .subscribe(
+                              answer -> {
+                                if (answer instanceof Result) {
+                                  Log.d(TAG, "setup an election");
+                                  openLaoDetail();
+                                } else {
+                                  Log.d(TAG, "failed to setup an election");
```

Comment:
I reckon this else block would be executed if we have an `Error` instance. Might be a good idea to log it

Reply:
Yes, I will then catch the error and log it